org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java

```diff
@@ -0,0 +1,255 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2024 Mountainminds GmbH & Co. KG and Contributors
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import static org.objectweb.asm.Opcodes.*;
+
+import org.jacoco.core.internal.instr.InstrSupport;
+import org.junit.Test;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.MethodNode;
+
+/**
+ * Unit tests for {@link KotlinComposeFilter}.
+ */
+public class KotlinComposeFilterTest extends FilterTestBase {
+
+	private final IFilter filter = new KotlinComposeFilter();
+
+	/**
+	 * <pre>
+	 * &#064;androidx.compose.runtime.Composable
+	 * fun example(x: Int) {
+	 *   if (x < 0)
+	 *     return
+	 *   example(x - 1)
+	 * }
+	 * </pre>
+	 *
+	 * transformed by <a href=
+	 * "https://github.com/JetBrains/kotlin/tree/v2.0.0-RC2/plugins/compose">Compose
+	 * Kotlin compiler plugin version 2.0.0</a> into
+	 *
+	 * <pre>
+	 * fun example(x: Int, $composer: Composer?, $changed: Int) {
+	 *   $composer = $composer.startRestartGroup(...)
+	 *   sourceInformation($composer, ...)
+	 *   val $dirty = $changed
+	 *   if ($changed and 0b0110 == 0) {
+	 *     $dirty = $dirty or if ($composer.changed(x)) 0b0100 else 0b0010
+	 *   }
+	 *   if ($dirty and 0b0011 != 0b0010 || !$composer.skipping) {
+	 *     if (isTraceInProgress()) {
+	 *       traceEventStart(...)
+	 *     }
+	 *
+	 *     if (x < 0) {
+	 *       if (isTraceInProgress()) {
+	 *         traceEventEnd()
+	 *       }
+	 *       $composer.endRestartGroup()?.updateScope { ... }
+	 *       return
+	 *     }
+	 *
+	 *     example(x, $composer, 0)
+	 *
+	 *     if (isTraceInProgress()) {
+	 *       traceEventEnd()
+	 *     }
+	 *   } else {
+	 *     $composer.skipToGroupEnd()
+	 *   }
+	 *   $composer.endRestartGroup()?.updateScope { ... }
+	 * }
+	 * </pre>
+	 */
+	@Test
+	public void should_filter() {
+		context.classAnnotations
+				.add(KotlinGeneratedFilter.KOTLIN_METADATA_DESC);
+		final MethodNode m = new MethodNode(InstrSupport.ASM_API_VERSION, 0,
+				"f", "(Landroidx/compose/runtime/Composer;I)V", null, null);
+		m.visitAnnotation("Landroidx/compose/runtime/Composable;", false);
+
+		final Range range1 = new Range();
+		final Range range2 = new Range();
+		final Range range3 = new Range();
+		final Range range4 = new Range();
+		final Range range5 = new Range();
+		final Range range6 = new Range();
+		final Range range7 = new Range();
+
+		m.visitVarInsn(ALOAD, 1);
+		range1.fromInclusive = m.instructions.getLast();
+		m.visitLdcInsn(new Integer(-974630231));
```

Comment:
```suggestion
		m.visitLdcInsn(Integer.valueOf(-974630231));
```
Avoid deprecation warning on newer JDKs.

Reply:
BTW this code was generated by ASMifier, so this fix of warning can be contributed to it -  https://gitlab.ow2.org/asm/asm/-/blob/ASM_9_7/asm-util/src/main/java/org/objectweb/asm/util/ASMifier.java#L1483 😉

Reply:
And here is PR for ASM: https://gitlab.ow2.org/asm/asm/-/merge_requests/422